### PR TITLE
GC unused disk dimension.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @vekterli or @toregge PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/filestorage/filestormetrics.cpp [hunk 1]
```diff
@@ -262,24 +262,16 @@ FileStorMetrics::FileStorMetrics(const LoadTypeSet&)
...
-        // disks that are not in use too.
-        std::ostringstream desc;
-        std::ostringstream name;
-        name << "disk_" << i;
-        desc << "Disk " << i;
-        disks[i] = std::make_shared<FileStorDiskMetrics>( name.str(), desc.str(), loadTypes, this);
-        sum.addMetricToSum(*disks[i]);
-        disks[i]->initDiskMetrics(loadTypes, numStripes, threadsPerDisk);
+    if (disk) {
+        throw vespalib::IllegalStateException("Can't initialize disk twice", VESPA_STRLOC);
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/filestorage/filestormetrics.cpp:268]: Consider replacing with `assert`

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — APPROVED** [review]
👍

**Review by toregge — APPROVED** [review]

File: storage/src/vespa/storage/persistence/filestorage/filestormetrics.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/filestorage/filestormetrics.cpp]: Fixed

**Pull request merged**